validator/src/validator.rs

```diff
@@ -628,12 +649,41 @@ impl<TNetwork: Network, TValidatorNetwork: ValidatorNetwork>
             }
         });
 
-        ParkingState {
+        ValidatorState::ParkingState {
             park_tx_hash: tx_hash,
             park_tx_validity_window_start: validity_start_height,
         }
     }
 
+    fn reactivate(&self, blockchain: &Blockchain) -> ValidatorState {
+        // TODO: Get the last view change height instead of the current height
+        let validity_start_height = blockchain.block_number();
+
+        let reactivate_transaction = TransactionBuilder::new_reactivate_validator(
+            &self.fee_key(),
+            self.validator_address(),
+            &self.signing_key(),
+            Coin::ZERO,
+            validity_start_height,
+            blockchain.network_id(),
+        )
+        .unwrap(); // TODO: Handle transaction creation error
+        let tx_hash = reactivate_transaction.hash();
+
+        let cn = self.consensus.clone();
+        tokio::spawn(async move {
+            if cn.send_transaction(reactivate_transaction).await.is_err() {
```

Comment:
Since the mempool txn prioritization PR is already merged, you should include this transaction with high priority, similar to how it is now done for the unpark transaction